src/components/search/UseSearch.jsx

```diff
@@ -1,11 +1,14 @@
 import { getClassesData } from '../../util/DataBaseRequests';
 import { useState, useEffect } from 'react';
+import { useSearchParams } from 'react-router-dom';
+
 const useSearch = () => {
   const [classes, setClasses] = useState([]);
   const [currentPage, setCurrentPage] = useState(1);
   const [totalPages, setTotalPages] = useState(1);
   const [search, setSearch] = useState('');
   const [category, setCategory] = useState('');
+  const [searchParams, setSearchParams] = useSearchParams();
   const [loading, setLoading] = useState(false);
 
   const fetchClasses = async (searchTerm = '', page = 1) => {
```

Comment:
Using `setLoading` to `true` before initiating the function can be redundant. It's often better to initially set the `loading` state to `false` and the after the successful request setting `loading` to `false`